lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndex.cpp

```diff
@@ -784,9 +785,25 @@ Fvp::PrimSelections MayaHydraSceneIndex::UfePathToPrimSelections(const Ufe::Path
     // Not the best implementation performance-wise, as ufeToDagPath converts
     // the UFE path to a string, then does a Dag path lookup with the string.
     constexpr bool isSprim = false; // Can't handle sprims as of 15-Aug-2023.
-    SdfPath primPath = GetPrimPath(UfeExtensions::ufeToDagPath(appPath), isSprim);
+
+    auto dagPath = UfeExtensions::ufeToDagPath(appPath);
+    SdfPath primPath = GetPrimPath(dagPath, isSprim);
     TF_DEBUG(MAYAHYDRALIB_SCENE_INDEX)
         .Msg("    mapped to scene index path %s.\n", primPath.GetText());
+  
+    //Check if this maya node has a special SdfPath associated with it, this is for custom or maya usd data producers scene indices.
+    //The class MhDataProducersMayaNodeToSdfPathRegistry does a mapping between Maya nodes and USD paths.
+    //The maya nodes registered in this class are used by data producers as a parent to all
+    //primitives. This class is used when the user selects one of these
+    //maya nodes to return the matching SdfPath so that all prims child of this maya node are
+    //highlighted.
+    MDagPath shapeDagPath(dagPath);
+    shapeDagPath.extendToShape();
+    const SdfPath matchingPath = FVP_NS::DataProducersNodeHashCodeToSdfPathRegistry::Instance().GetPath(MObjectHandle(shapeDagPath.node()).hashCode());
```

Comment:
The usage of the DataProducersNodeHashCodeToSdfPathRegistry makes me feel we can unify with Fvp::PathMapperRegistry.